src/easylink/step.py

```diff
@@ -890,14 +912,16 @@ def set_configuration_state(
             self.step_graph.add_node_from_step(self.template_step)
             # Update the slot mappings with renamed children
             input_mappings = [
-                InputSlotMapping(slot, self.name, slot) for slot in self.input_slots
+                InputSlotMapping(slot, self.template_step.name, slot)
```

Comment:
this doesn't really matter since we set the template name to self.name above anyway, but seems slightly more accurate

Reply:
In this part of the if/else, we are generating the stepgraph from scratch and slot mappings from scratch. I'm sure there's a way to refactor here.

Reply:
Right, I understand that, but we then _also_ generate them from scratch in `_update_slot_mappings`. So when is this version even used?

Reply:
Yeah I see what you mean. _update_slot_mappings is called when the `TemplatedStep` is actually a templated step, i.e. it has the config key ("iterate" or "parallel") in the pipeline yaml. This entire first part of the if/else is basically special-handling the case where the user opts NOT to take advantage of templating.